src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs

```diff
@@ -512,6 +512,14 @@ private IControl ScrollIntoViewCore(int index)
             var generator = Owner.ItemContainerGenerator;
             var newOffset = -1.0;
 
+            if (!panel.IsMeasureValid && panel.PreviousMeasure.HasValue)
+            {
+                //before any kind of scrolling we need to make sure panel measure is valid
+                //or we risk get panel into not valid state
+                //we make a preemptive quick measure so scrolling is valid
+                panel.Measure(panel.PreviousMeasure.Value);
```

Comment:
Not sure if this should be a full layout here (i.e. `LayoutManager.ExecuteLayoutPass()` but if this seems to work in your testing I'm 👍 with doing it like this until we have evidence to the contrary.

Reply:
i wanted to be as minimal layout update is possible, only if needed,
as i'm bit afraid of full layout cycle (also few lines below), as in theory it possible `LayoutManager.ExecuteLayoutPass()` trigger
again `ScrollIntoView `and thus cause stackoverflow, we might introduce a flag to prevent layoutcycle for this situation for this method what do you think?